### DISCUSSION TITLE
I want to make 2 Process Synchronization in C# with Mutex

### DISCUSSION THREAD

**Discussion by bharatbodage** [dotnet/runtime#62041] [Q&A]
I have requirement where we have 2 exe's. When first exe started then I have to call method of second exe which will do some process. once that process is completed then it again intimates first exe that my work is done. you can resume your work.

Note: I do not want to use sockets. Most preferred way is like Mutex or Semaphore.

Any help will be much appreciated specifically with some example

Following is the actual workflow which I need to handle with Mutex.
![ProcessWorkflow](https://user-images.githubusercontent.com/15664712/143409328-26aa10fa-a706-44f7-b383-553341bb1724.png)

**Comment by gfoidl** [comment]
> When first exe started then I have to call method of second exe

Who starts the seconds one? 
If it's the first exe, you could use [`Process.WaitForExit`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit).

If not the first one starts the second one, how do you intend to start it? I see "no sockets", so there are further ways for IPC available, but please answer the question first.

**Reply by bharatbodage** [reply]
Second process is started by user and not started by first process. They both are different processes.

**Reply by gfoidl** [reply]
Yould could use _named pipes_ for the IPC.
So the first process can wait for a "pipe response" from the second process.

**Reply by bharatbodage** [reply]
Can you please give coding example for named pipes?
